app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -729,6 +731,75 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml): void
         }
     }
 
+    /**
+     * Replaces strings the PHP date string parser understands with
+     * the actual time in date, datetime and hidden fields.
+     *
+     * Hidden fields must end with `|date` to prevent replacing
+     * strings by a date time representation by accident.
+     */
+    public function populateDateTimeValues(Form $form, string &$formHtml): void
+    {
+        $formName       = $form->generateFormName();
+        $fields         = $form->getFields();
+        $autoFillFields = [];
+
+        /** @var Field $field */
+        foreach ($fields as $key => $field) {
+            if (in_array($field->getType(), ['date', 'datetime', 'hidden'])) {
+                $autoFillFields[$key] = $field;
+            }
+        }
+
+        foreach ($autoFillFields as $field) {
+            $fieldType = $field->getType();
+            $token     = $field->getDefaultValue();
+
+            switch ($fieldType) {
+                case 'date':
+                    $format = 'Y-m-d';
+                    break;
+                case 'datetime':
+                case 'hidden':
+                    $format = 'Y-m-d\TH:i';
+                    break;
+
+                default:
+                    continue 2;
+            }
+
+            // prevent empty fields from getting parsed as now
+            if (empty(trim($token))) {
+                continue;
+            }
+
+            if ('hidden' === $fieldType) {
+                /* $pattern:
+                        ^([^\|]+) => match any char, except pipe
+                        \|date    => then look for `|date`
+                */
+                preg_match('/^([^\|]+)\|date$/', $token, $matches);
```

Comment:
Would this simpler regex work? Why do we need to exclude pipes?
```suggestion
                preg_match('/(.*)\|date$/', $token, $matches);
```

Reply:
I once read a piece about how `.*` might lead to issues with your regex, especially if it is/becomes ungreedy. Thus, I avoid using `.*`, where possible. `[^\|]` is basically the same, just looks a bit unfamiliar.

I know that PHP's regex is greedy by default, so `.*` isn't a problem per se. This just felt more comfortable to me. I can change it, however.

Reply:
It sounds like you know what you are doing. No need to change it. It was already a nitpick. The only concern from my side was readability. I prefer simpler solutions. All good here.